src/Tasks/GenerateResource.cs

```diff
@@ -2432,6 +2432,12 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                     {
                         if (String.Compare(requestedAssemblyName.Name, candidateAssemblyName.Name, StringComparison.CurrentCultureIgnoreCase) == 0)
                         {
+                            string itemSpec = _assemblyFiles[i].ItemSpec;
+
+                            if (requestedAssemblyName.Name.Contains("System.Runtime"))
```

Comment:
/cc @weshaggard 
I suspect you're getting that error when you cross type-universes.  IIRC reflection-only-load doesn't handle loading different core assemblies than what your running on.

I think the right fix here is to *stop using ReflectionOnlyLoad*.  Switch to something like System.Reflection.Metadata.  It looks like your usage of reflection is minimal so that might even speed up your code here.  Alternatively you could try to get CLR to fix ReflectionOnlyLoad with other core assembleis but I'm guessing that even if a fix was taken it would take a long time to make it to desktop where you could depend on it.

This hack is replacing application targeted assemblies with build-time assemblies, so I cannot really sign off on it as a safe thing to do.  With that caveat I will try and answer your questions.
1. I believe you're only working around the core-assembly issue, so simply using the System.Runtime facade instead of reference may be sufficient.
2.  Perhaps you should just try to load it by name and let the build runtime use whatever rules it has to (binding-redirect/unification/tpa-list) to locate the right one for the build framework.